pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/utilities/SecuredHDFS.java

```diff
@@ -53,6 +53,10 @@
     public static void verifyToken(ProtocolData protData, ServletContext context) {
         try {
             if (UserGroupInformation.isSecurityEnabled()) {
+                UserGroupInformation loginUser = UserGroupInformation.getLoginUser();
+                if (!loginUser.hasKerberosCredentials()) {
```

Comment:
I don't think so if we pass the above check `UserGroupInformation.isSecurityEnabled()` but I can add a null check to be safe